### PR TITLE
Implement registerContract, registerFunction, and listContracts

### PR DESCRIPTION
**Pull request by supl**
This PR implements three related functions in the ClientServiceBase class.

- registerContract (from https://github.com/scalar-labs/scalardl-javascript-sdk-base/blob/master/index.js#L218)

- registerFunction (from https://github.com/scalar-labs/scalardl-javascript-sdk-base/blob/master/index.js#L158)

- listContracts (from https://github.com/scalar-labs/scalardl-javascript-sdk-base/blob/master/index.js#L282)

### CODE REVIEW

File: node/common/builder/contract_listing_request.ts [hunk 1]
```diff
@@ -1,12 +1,6 @@
 import {TextEncoder} from '../polyfill/text_encoder';
 import {SignatureSigner} from '../signature';
-
-export type ContractsListingRequest = {
-  setCertHolderId: (id: string) => void;
-  setCertVersion: (version: number) => void;
-  setContractId: (id: string) => void;
-  setSignature: (signature: Uint8Array) => void;
-};
+import {ContractsListingRequest} from '../scalar.proto';
```
Comment (supl) [inline node/common/builder/contract_listing_request.ts:4]: Move the type definition to the scalar.proto.ts file.
(scalar.proto.ts defines the types we need in TypeScript for Protocol Buffers compiler generated files)

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: node/common/client_service_base.ts [hunk 2]
```diff
@@ -169,4 +353,71 @@ export class ClientServiceBase {
...
+      .withCertVersion(this.config.getCertVersion())
+      .build();
+  }
+
+  private createSigner(): SignatureSigner {
+    const config = this.config;
+
+    if (!config.getPrivateKeyCryptoKey() && config.getPrivateKeyPem() === '') {
+      throw new Error(
+        'Eitehr private key crypto key or private key pem must be set'
```
Comment (jnmt) [inline node/common/client_service_base.ts]:
```suggestion
        'Either private key crypto key or private key pem must be set'
```

**Review by jnmt — COMMENTED** [review]
Overall looks good, thank you! I just left a minor suggestion.

**Comment by supl** [comment]
> Overall looks good, thank you! I just left a minor suggestion.

Thank you for the catch!

**Review by jnmt — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**